pkg/broker/ingress/handler_test.go

```diff
@@ -0,0 +1,79 @@
+package ingress
+
+import (
+	"context"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client/test"
+	"github.com/google/go-cmp/cmp"
+)
+
+// TestHandlerChannelInbound uses a channel based mock client for both inbound and decouple.
+func TestHandlerChannelInbound(t *testing.T) {
+	inbound, ic := test.NewMockReceiverClient(t, 1)
+	decouple, dc := test.NewMockSenderClient(t, 1)
+	createAndStartIngress(t, inbound, decouple)
+
+	input := createTestEvent()
+	// Send an event to the inbound receiver client.
+	ic <- input
+	// Retrieve the event from the decouple sink.
+	output := <-dc
+
+	if dif := cmp.Diff(input, output); dif != "" {
+		t.Errorf("Output event doesn't match input, dif: %v", dif)
+	}
+}
+
+// TestHandlerDefaultHTTPInbound uses default HTTP inbound client and a channel based mock decouple
+// sink.
+func TestHandlerDefaultHTTPInbound(t *testing.T) {
+	decouple, dc := test.NewMockSenderClient(t, 1)
+	createAndStartIngress(t, nil, decouple)
+
+	input := createTestEvent()
+	// Send an event to the inbound receiver client.
+	sendEventOverHTTP(t, "http://localhost:8080", input)
+	// Retrieve the event from the decouple sink.
+	output := <-dc
+
+	if dif := cmp.Diff(input, output); dif != "" {
+		t.Errorf("Output event doesn't match input, dif: %v", dif)
+	}
+}
+
+func sendEventOverHTTP(t *testing.T, url string, events ...cloudevents.Event) {
+	p, err := cloudevents.NewHTTP(cloudevents.WithTarget(url))
+	if err != nil {
+		t.Fatalf("Failed to create HTTP protocol: %+v", err)
+	}
+	ce, err := cloudevents.NewClient(p)
+	if err != nil {
+		t.Fatalf("Failed to create HTTP client: %+v", err)
+	}
+	for _, event := range events {
+		ce.Send(context.Background(), event)
+	}
+}
+
+// createAndStartIngress creates an ingress and calls its Start() method in a goroutine.
+func createAndStartIngress(t *testing.T, inbound cloudevents.Client, decouple DecoupleSink) *handler {
+	ctx := context.Background()
```

Comment:
Does this need to get closed? If so, I recommend returning a 'cleanup' goroutine. So the function's return would become something like  `(*hander, func())` and the caller defers calling the clean up function.

Reply:
[T.Cleanup](https://golang.org/pkg/testing/#T.Cleanup) makes this easy.

Reply:
I couldn't find any `cleanUp` methods in the underlying clients so I guess no need to do so.

Reply:
You're right, looking through the code it doesn't need to happen because the context is only used to create the decoupler, which is passed in as an option here. 

My preference is to add a comment or create the clean up method anyway, but I won't hold up the PR for it.

Reply:
SG. Added a placeholder cleanup function.